src/js/entities-service/patients.js

```diff
@@ -9,12 +11,28 @@ const Entity = BaseEntity.extend({
     'patients:model': 'getModel',
     'fetch:patients:model': 'fetchModel',
     'fetch:patients:model:byAction': 'fetchPatientByAction',
+    'fetch:patient:engagementStatus': 'fetchPatientEngagementStatus',
+    // 'fetch:patient:engagementSettings': 'fetchPatientEngagementSettings',
   },
   fetchPatientByAction(actionId) {
     const patient = Radio.request('entities', 'patients:model');
 
     return patient.fetch({ url: `/api/actions/${ actionId }/patient` });
   },
+  fetchPatientEngagementStatus(patientId) {
+    const patient = Radio.request('entities', 'patients:model', patientId);
+    const engagement = patient.getEngagement();
+
+    return $.ajax({
+      url: `/api/patients/${ patientId }/engagement-status`,
+    })
+      .then(response => {
+        return patient.set({
+          engagement: _.extend(engagement, response.data),
```

Comment:
extending the original engagement data means the set already happened.  I think `patient.set(response.data)` should be ok here.